0049-Group-Anagrams.py

```diff
@@ -0,0 +1,25 @@
+'''
+Given an array of strings, group anagrams together.
+
+Example:
+
+Input: ["eat", "tea", "tan", "ate", "nat", "bat"],
+Output:
+[
+  ["ate","eat","tea"],
+  ["nat","tan"],
+  ["bat"]
+]
+
+Note:
+
+    All inputs will be in lowercase.
+    The order of your output does not matter.
+'''
+class Solution:
```

Comment:
time and space complexity?

Reply:
Time Complexity: O(n*k)
Space Cimplexity: O(1) if considering hash table as an output DS

Reply:
1. Here the hash table is not an output DS, the output is in array form.
2. I don't think so hashing is a good approach here. As dictionary also do hashing on its keys. So, in this case, your string gets hashed two times.

Reply:
okay, will get back to you